plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTable.java

```diff
@@ -13,203 +13,55 @@
-            @JsonProperty("queryOptions") Map<String, String> queryOptions,
-            @JsonProperty("query") String query)
-    {
-        this.tableName = requireNonNull(tableName, "tableName is null");
-        this.suffix = requireNonNull(suffix, "suffix is null");
-        this.projections = ImmutableList.copyOf(requireNonNull(projections, "projections is null"));
-        this.filter = requireNonNull(filter, "filter is null");
-        this.groupingColumns = ImmutableList.copyOf(requireNonNull(groupingColumns, "groupingColumns is null"));
-        this.aggregateColumns = ImmutableList.copyOf(requireNonNull(aggregateColumns, "aggregateColumns is null"));
-        this.havingExpression = requireNonNull(havingExpression, "havingExpression is null");
-        this.orderBy = ImmutableList.copyOf(requireNonNull(orderBy, "orderBy is null"));
-        this.limit = requireNonNull(limit, "limit is null");
-        this.offset = requireNonNull(offset, "offset is null");
-        this.queryOptions = ImmutableMap.copyOf(requireNonNull(queryOptions, "queryOptions is null"));
-        this.query = requireNonNull(query, "query is null");
-        this.isAggregateInProjections = projections.stream()
-                .anyMatch(PinotColumnHandle::isAggregate);
-    }
-
-    @JsonProperty
-    public String getTableName()
-    {
-        return tableName;
-    }
-
-    @JsonProperty
-    public Optional<String> getSuffix()
-    {
-        return suffix;
-    }
-
-    @JsonProperty
-    public List<PinotColumnHandle> getProjections()
-    {
-        return projections;
-    }
-
-    @JsonProperty
-    public Optional<String> getFilter()
-    {
-        return filter;
-    }
-
-    @JsonProperty
-    public List<PinotColumnHandle> getGroupingColumns()
-    {
-        return groupingColumns;
-    }
-
-    @JsonProperty
-    public List<PinotColumnHandle> getAggregateColumns()
+    public DynamicTable
     {
-        return aggregateColumns;
+        requireNonNull(tableName, "tableName is null");
+        requireNonNull(suffix, "suffix is null");
+        projections = ImmutableList.copyOf(requireNonNull(projections, "projections is null"));
+        requireNonNull(filter, "filter is null");
+        groupingColumns = ImmutableList.copyOf(requireNonNull(groupingColumns, "groupingColumns is null"));
+        aggregateColumns = ImmutableList.copyOf(requireNonNull(aggregateColumns, "aggregateColumns is null"));
+        requireNonNull(havingExpression, "havingExpression is null");
+        orderBy = ImmutableList.copyOf(requireNonNull(orderBy, "orderBy is null"));
+        requireNonNull(limit, "limit is null");
+        requireNonNull(offset, "offset is null");
+        queryOptions = ImmutableMap.copyOf(requireNonNull(queryOptions, "queryOptions is null"));
+        requireNonNull(query, "query is null");
     }
 
-    @JsonProperty
-    public Optional<String> getHavingExpression()
+    public boolean aggregateInProjections()
```

Comment:
Renamed the method from `isAggregateInProjections` and moved the logic from a constructor.